### ISSUE TITLE
`Vision Language Model`

### ISSUE THREAD

**Issue by xp1632** [xp1632/Aalen_working_log#21]
### How does the vision language model work?

- After reading this paper, you'll find the answer:
- LLaVa:  Large Language and Vision Assistant Visual Instruction Tuning
https://llava-vl.github.io/
- paper2: Improved Baselines with Visual Instruction Tuning 

---

- Before reading this paper, I know that, in general, they treat `image` as a type of `foreign language`
- the image is divided by grids and image feature patches are also tokenized in an `image encoder` just like the `text encoder` in text LLM

---

### How to produce the dataset for fine-tuning?

- In version `1.0 of LLaVa`: - paper1: Visual Instruction Tuning https://arxiv.org/pdf/2304.08485
-  they also mentioned how they built **a pipeline to generate the fine-tuning dataset** they required
-
- In `Cambrian-1`(https://cambrian-mllm.github.io/), they find out the dataset they collected from the public source or prompting to GPT doesn't meet their requirements
- -
- Before reading the paper, I know:
- They first build a small dataset and a small model, generate a slightly larger dataset, then introduce human intervention to improve the dataset then the small model, and get a 10% `golden dataset`
- Then based on this `10% golden dataset`, they generate the rest 90% dataset based on their data pipeline

---

### Except for the end-end generation, to avoid hallucination, is there the `white-box` GPT that we can also see the process in-between such as `code generation`?

- the previously mentioned MLLM are black boxed, which means we lose the information of `how` this model generates the answer
- to avoid hallucination which is quite common for generative models
- `ViperGPT` (https://viper.cs.columbia.edu/) also gives transparent reasoning of how this answer is generated in the white box

 
![viper_gpt](https://github.com/user-attachments/assets/0b014018-dca8-4379-a33a-644a3a1ab56a)

- In ViperGPT demo, we noticed that the reasoning for the answer is demonstrated with both python code and its respective visual inference

**Comment by xp1632** [comment]
### Paper to read:

1. **Segment Anything**: 
- paper: https://arxiv.org/pdf/2304.02643
- important blog: https://ai.meta.com/blog/segment-anything-foundation-model-image-segmentation/
- code: https://github.com/facebookresearch/segment-anything 
- **Questions**: 
- how did they prepare the dataset/mask?   
- what's a `promptable segmentation`? : 
- how did they deal with the segmentation prompts, especially the combination of additional markers, and semantic text

---

2. **LLaVA: Large Language and Vision Assistant Visual Instruction Tuning**:
- project: https://llava-vl.github.io/
- paper1: Visual Instruction Tuning https://arxiv.org/pdf/2304.08485
- paper2: Improved Baselines with Visual Instruction Tuning https://arxiv.org/pdf/2310.03744
- **Questions**: 
- how does the visual language model work?
- what's their standard/structure of dataset for finetuning?
- what does their proposed pipeline to generate the dataset look like?


---

3. **Cambrian(寒武纪)-1: A Fully Open, Vision-CentricExploration of Multimodal LLMs**
- project: https://cambrian-mllm.github.io/ 
- paper: https://arxiv.org/pdf/2406.16860
- dataset: https://huggingface.co/datasets/nyu-visionx/Cambrian-10M
- **Questions**: 
- what's a multimodal llm?
- how does the dataset look like to train a mllm?
- how to deal with the `not-enough-training data` problem(by generating more data based on `golden` ones)?
- how to build a `golden dataset` as the seed for generating the rest datasets?
- how do you know if the current dataset is `golden` or not?

---

4. **ViperGPT: Visual Inference via Python Execution for Reasoning**
- project: https://viper.cs.columbia.edu/
- paper: https://arxiv.org/pdf/2303.08128
- code: https://github.com/cvlab-columbia/viper
- **Questions**: 
- how did vipergpt link the python code and visual inference?
- what does their training dataset look like, is there also `code snippet` component in it?
- Answer after 1st reading: https://github.com/xp1632/Aalen_working_log/issues/3

**Comment by xp1632** [comment]
### The two previous comment are from our summary in 
https://github.com/xp1632/DFKI_working_log/issues/75

- while vipergpt's environment too dirty, we'll start from one single model 

----

### Our aiming VLM list is:


- Microsoft's BLIP-2
-  ViT (Vision Transformer) - Simple but effective
-  CLIP - Good for general image understanding
- Glip - the one that vipergpt used 
- other models in Huggingface: https://huggingface.co/docs/transformers/en/model_doc/blip-2
